src/main/java/duke/task/DeadLine.java

```diff
@@ -0,0 +1,51 @@
+package duke.task;
+
+import duke.dukeexception.DukeException;
+
+/**
+ * A class represents task with deadline.
+ */
+public class DeadLine extends Task {
+    private String by;
+    private String description;
+
+    /**
+     * A class constructor
+     * @param description Full version of the input
+     * @throws DukeException Throws when error occurs
+     */
+    public DeadLine(String description) throws DukeException {
+        super("tempTask");
+        try {
+            String temp=description.split(" ")[1];
+        } catch (IndexOutOfBoundsException ie) {
+            throw (new DukeException("OOPS!!! The description of a deadline cannot be empty."));
+        } try {
+            super.correctDescrition(description.split("by")[0].split(" ", 2)[1]);
```

Comment:
Abit long and complicated to read. Might be better breaking up into separate lines.